pkg/es/config/config.go

```diff
@@ -362,6 +363,18 @@ func (c *Configuration) getConfigOptions(logger *zap.Logger) ([]elastic.ClientOp
 	}
 	options = append(options, elastic.SetHttpClient(httpClient))
 	options = append(options, elastic.SetBasicAuth(c.Username, c.Password))
+
+	// Elastic client requires a "Printf"-able logger.
+	l := zapgrpc.NewLogger(logger)
+	switch {
+	case logger.Core().Enabled(zap.DebugLevel):
+		l = zapgrpc.NewLogger(logger, zapgrpc.WithDebug())
+		options = append(options, elastic.SetTraceLog(l))
+	case logger.Core().Enabled(zap.InfoLevel):
```

Comment:
I am not sure this is a good idea. Our default log level is info, so we're going to be adding a log line for every span written in the collector.

Reply:
Okay, fair point about the collector. The motivation for this PR was to log ES queries so I think just showing the tracelog when `debug` logging is enabled is enough, and the rest, including info level in jaeger, should default to `ErrorLevel` logging in ES client.

It also brings up the question of whether the debug logs for collector would be too much as it would log the entire span's content which can be very large. The only log line the collector writes when `log-level=debug` is:
```
{"level":"debug","ts":1615007007.183977,"caller":"app/span_processor.go:149","msg":"Span written to the storage by the collector","trace-id":"4a9afb787bbb4f3f","span-id":"4a9afb787bbb4f3f"}
```

which I can see a use case for when debugging ingestion. With ES queries in the logging mix, it may be hard to see the above logs without grepping. As such, maybe we need to isolate this feature to jaeger-query and not jaeger-collector?

Reply:
You're probably better positioned to answer this since you're running this setup in prod.

Another option instead of piggy-backing on the main log level is to add just an `--es.` flag for log level, then people can go nuts if they want to, all the way to full payload.

Reply:
Actually, we don't use jaeger-collector with ES as the backing store; although I do like your suggestion of having a separate flag (I propose `--es.log-level`), which I'll aim to do in the coming day or so.

Reply:
@yurishkuro I've added the `--es.log-level` (and `--es-archive.log-level`) flag and tested it together with `--log-level` in different combinations of values. Both the `--es` and non-`--es` flags are decoupled, working independently of one another.